exp-models/tests/unit/utils/csv-writer-test.js

```diff
@@ -0,0 +1,160 @@
+import { NL, squash, uniqueFields, writeCSV} from 'exp-models/utils/csv-writer';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | csv writer');
+
+const one = 'one';
+const two = 'two';
+const three = 'three';
+const four = 'four';
+const five = 'five';
+const six = 'six';
+const seven = 'seven';
+
+test('squash', function(assert) {
+    assert.deepEqual(
+        squash({
+            alpha: one,
+            bravo: two,
+            charlie: {
+                delta: three
+            },
+            echo: {
+                foxtrot: {
+                    golf: four
+                }
+            },
+            hotel: {
+                india: five,
+                juliet: {
+                    kilo: six
+                }
+            },
+            lima: {
+                mike: {
+                    november: {
+                        oscar: {
+                            papa: {
+                                quebec: seven
+                            }
+                        }
+                    }
+                }
+            }
+        }),
+        {
+            alpha: one,
+            bravo: two,
+            'charlie.delta': three,
+            'echo.foxtrot.golf': four,
+            'hotel.india': five,
+            'hotel.juliet.kilo': six,
+            'lima.mike.november.oscar.papa.quebec': seven
+        }
+    );
+});
+
+test('squash with map', function(assert) {
+    assert.deepEqual(
+        [
+            {
+                alpha: one,
+                bravo: {
+                    charlie: one
+                }
+            },
+            {
+                alpha: two,
+                bravo: {
+                    charlie: two
+                }
+            },
+            {
+                alpha: three,
+                bravo: {
+                    charlie: three
+                }
+            }
+        ].map(squash),
+        [
+            {
+                alpha: one,
+                'bravo.charlie': one
+            },
+            {
+                alpha: two,
+                'bravo.charlie': two
+            },
+            {
+                alpha: three,
+                'bravo.charlie': three
+            }
+        ]
+    );
+});
+
+test('uniqueFields', function(assert) {
+    assert.deepEqual(
```

Comment:
If we do go ahead with using unique fields: do we need to consider nesting? Or do we simply always assume that `uniqueFields` will already be squashed?